Feature/pidf controllers

Ursa-Major22774 opened the pull request:
Before issuing a pull request, please see the contributing page.

BeepBot99 commented:
You should probably put it in `org.firstinspires.ftc.teamcode.pedroPathing`

BeepBot99 commented:
> it's safer this way for compatibility but i get what you're saying, you can put anything as the last package including pedroPathing

How is it safer this way? It is normal to put code in subdirectories of teamcode. Additionally, dairy still works with it like that.

minusxne commented:
im applying the same logic as baron, you can just rename the package, you could put anything there including pedroPathing but you're not risking any compatibility by just doing it yourself instead of it coming named by default but it works even if you don't want to rename it

BeepBot99 commented:
> im applying the same logic as baron, you can just rename the package, you could put anything there including pedroPathing but you're not risking any compatibility by just doing it yourself instead of it coming named by default but it works even if you don't want to rename it

Yes, but it is standard for library code to go in a seperate directory.

minusxne commented:
> > im applying the same logic as baron, you can just rename the package, you could put anything there including pedroPathing but you're not risking any compatibility by just doing it yourself instead of it coming named by default but it works even if you don't want to rename it
> 
> Yes, but it is standard for library code to go in a seperate directory.

true, since the original problem is a non standard package there should be another pedro package under the teamcode one to respect the standard
i'll redo the pr and correct some other unnecessary lines changed too like imports reformatted by android studio

BeepBot99 commented:
In the future you can just update the existing pr instead of creating a new one.

minusxne commented:
yes but because the changes are substantial it calls for a new one, it's just a good devOps practice.

rowan-mcalpin commented:
It doesn't really matter but in this case since the changes are entirely focused on one thing, they don't really count as "significant" even if a lot of files changed.

minusxne commented:
changing the package is significant enough that you wouldn't leave 2 commits with 2 separate package names in a pr focused on the initial one. I get what you're trying to say but it's not that important. I get that some people have other workflows on git but it's not a situation of right and wrong, just preference. Can we please focus on the issue at hand since i feel like this is going nowhere and my team and others want to have dairy support in the quickstart and not have to worry about package higherarchy.

minusxne commented:
To make things clearer moving forward, I'm happy to consolidate any changes into less prs to allign with your specific git practices even if the changes can be interpreted as major in certain ways and I would like to keep things professional and not beat arround the bush with this problem even if it is a trivial one. I will keep suggestions relating to ettiqete in mind moving forward if anything else comes up.

BaronClaps committed:
2.0.0

BaronClaps committed:
2.0.0 - Template Constants

BaronClaps committed:
2.0.0 - Template Constants

BaronClaps committed:
Add Circle to the selector

google-labs-jules[bot] committed:
feat: Add production-quality PIDF controllers

Creates a set of production-quality, reusable PIDF controllers for FIRST Tech Challenge robots.

- `LinearPIDFController`: For controlling linear mechanisms like slides. Features PID, feedforward (kS, kV), derivative-on-measurement, anti-windup, output clamping, low-pass filtering, and optional voltage compensation.

- `RotationalPIDFController`: Extends the linear controller to add a gravity feedforward term (kG), making it suitable for arms and other rotational mechanisms.

- `VelocityPIDFController`: A dedicated controller for attaining and maintaining a specific motor velocity.

- `UsageExample`: A simple class to demonstrate the instantiation, configuration, and use of each controller.

All controllers use internal timing, are unit-agnostic, and do not depend on any FTC SDK classes, ensuring maximum reusability.

google-labs-jules[bot] committed:
docs: Add tuning guide for PIDF controllers

Adds a `tuning.txt` file to the `PIDFController` directory.

This file provides a comprehensive, step-by-step guide on how to tune the gains (`kP`, `kI`, `kD`, `kS`, `kV`, `kG`) for all three controller classes. It outlines a systematic tuning methodology, explains the purpose of each gain, and provides practical tips for achieving a stable and responsive system.

Ursa-Major22774 committed:
Merge branch 'master' into feature/pidf-controllers
